### PR TITLE
Fix for MySQL multi-host internally_generated connection urls

### PR DESCRIPTION
**Pull request by ygree**
# What Does This Do

Fixes a reported problem when the instrumentation failed to parse MySQL multi-host connection URLs such as `"jdbc:mysql:loadbalance://**internally_generated**1711497942886**"`

Uses an original connection URL to extract database information and attaches it to the underlying connection object so that it can be properly propagated and used with and without the split-by-instance setting.

# Motivation

Provide a fix for a reported bug to properly extract and store database connection information from the MySQL multi-host connection URL and prevent a parsing exception.

# Additional Notes

Jira ticket: [APMS-11717]

[APMS-11717]: https://datadoghq.atlassian.net/browse/APMS-11717?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Comment by am312** [comment]
This might be good to eventually include in a broader solution covering the variety of unsupported connection URLs I discussed here:
https://docs.google.com/document/d/1m5unhxc4GWwcTXr1za1pPoOQexSlTyWZrXzOqcwWho0/edit?usp=sharing

File: dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/jdbc/JDBCConnectionUrlParser.java [hunk 1]
```diff
@@ -824,7 +830,7 @@ public static DBInfo parse(String connectionUrl, final Properties props) {
       }
       return GENERIC_URL_LIKE.doParse(connectionUrl, parsedProps).build();
     } catch (final Exception e) {
-      ExceptionLogger.LOGGER.debug("Error parsing URL", e);
+      ExceptionLogger.LOGGER.debug("Error parsing URL {}", jdbcUrl, e);
```
Comment (mcculls) [inline dd-java-agent/agent-bootstrap/src/main/java/datadog/trace/bootstrap/instrumentation/jdbc/JDBCConnectionUrlParser.java:833]: The full URL could contain sensitive details - can we merge the fix first and discuss this logging change separately?

**Review by mcculls — APPROVED** [review]
LGTM, but I'd leave the logging change to a separate PR/discussion

**Comment by ygree** [comment]
> This might be good to eventually include in a broader solution covering the variety of unsupported connection URLs I discussed here: https://docs.google.com/document/d/1m5unhxc4GWwcTXr1za1pPoOQexSlTyWZrXzOqcwWho0/edit?usp=sharing

This change doesn't seem to be relevant to the broader initiative, since it's not about supporting a different connection URL, but about properly handling connection proxy objects.

**Pull request merged**